src/components/ListView.js

```diff
@@ -53,22 +89,29 @@ const ListView = ({ shoppingList, loading, error }) => {
           </form>
 
           <ul className="list">
-            {shoppingList &&
-              shoppingList[0] &&
-              shoppingList[0].items.map((item) => {
-                const searchResult = item.name.includes(
-                  value.toLowerCase().trim(),
-                );
-                return (
-                  searchResult && (
-                    <ListItem
-                      key={item.id}
-                      item={item}
-                      shoppingList={shoppingList}
-                    />
-                  )
-                );
-              })}
+            {sortedList.map((item) => {
+              const searchResult = item.name.includes(
+                value.toLowerCase().trim(),
+              );
+              //pass parameters to checkIndex function
+              const lastPurchase = item.LastPurchasedDate;
```

Comment:
Instead of introducing 3 extra lines (91-94), why not pass ``item.LastPurchasedDate``, ``item.daysLeftForNextPurchase``,
``item.numberOfPurchases`` straight away to the checkIndex function?
 
Also, your ``LastPurchasedDate`` starts with capslock

Reply:
Felt it was more readable that way, but it's okay, I'll change it, thanks.
Thanks for the catch, the typo was causing major problems!